src/platform/forms-system/src/js/containers/FormPage.jsx

```diff
@@ -110,6 +113,38 @@ class FormPage extends React.Component {
     this.props.router.push(path);
   };
 
+  prePopulateArrayData = () => {
+    const { pageConfig } = this.props.route;
+    // only applicable to array routes with these settings
+    if (pageConfig.showPagePerItem && pageConfig.allowPathWithNoItems) {
+      const arrayFormData = get(
+        [pageConfig.arrayPath, this.props.params.index],
+        this.props.form.data,
+      );
```

Comment:
This code is also in the `onChange` - should it get removed, or maybe make a reusable utility function that performs this check?

Reply:
good idea, there was a lot of repeat code in this file, so I refactored and created get and set methods for this specific data.